File: app/src/main/java/br/com/onibusapp/onibusapp/adapter/FavoritoViewHolder.java
package br.com.onibusapp.onibusapp.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import br.com.onibusapp.onibusapp.R;
import br.com.onibusapp.onibusapp.data.dominio.Favorito;

/**
 * Created by diego on 24/07/2018.
 */

public class FavoritoViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnLongClickListener {

    private RecyclerViewOnClickListener recyclerViewOnClickListener;
    private TextView txtSentido;
    private TextView txtLinha;

    public FavoritoViewHolder(View itemView, RecyclerViewOnClickListener recyclerViewOnClickListener) {
        super(itemView);
        this.txtSentido = itemView.findViewById(R.id.txt_sentido);
        this.txtLinha = itemView.findViewById(R.id.txt_linha);
        this.recyclerViewOnClickListener = recyclerViewOnClickListener;
        itemView.setOnClickListener(this);
        itemView.setOnLongClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if (this.recyclerViewOnClickListener != null)
            this.recyclerViewOnClickListener.onClickListener(v, getPosition());
    }

    @Override
    public boolean onLongClick(View v) {
        if (this.recyclerViewOnClickListener != null)
            this.recyclerViewOnClickListener.onLongPressClickListener(v, getLayoutPosition());
        return false;
    }

    public void setFavorito(Favorito favorito) {
        txtSentido.setText(favorito.getCodigoSentido().toString());
        txtLinha.setText(favorito.getNomeLinha());
    }
}

File: app/src/main/java/br/com/onibusapp/onibusapp/adapter/FavoritoViewHolder.kt
package br.com.onibusapp.onibusapp.adapter

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

import br.com.onibusapp.onibusapp.R
import br.com.onibusapp.onibusapp.data.dominio.DestinoEnum
import br.com.onibusapp.onibusapp.data.dominio.Favorito

/**
 * Created by diego on 24/07/2018.
 */

class FavoritoViewHolder(itemView: View, private val recyclerViewOnClickListener: RecyclerViewOnClickListener?) : RecyclerView.ViewHolder(itemView), View.OnClickListener, View.OnLongClickListener {

    private val txtSentido: TextView
    private val txtLinha: TextView

    init {
        this.txtSentido = itemView.findViewById(R.id.txt_sentido)
        this.txtLinha = itemView.findViewById(R.id.txt_linha)
        itemView.setOnClickListener(this)
        itemView.setOnLongClickListener(this)
    }

    override fun onClick(v: View) {
        if (this.recyclerViewOnClickListener != null)
            this.recyclerViewOnClickListener.onClickListener(v, position)
    }

    override fun onLongClick(v: View): Boolean {
        if (this.recyclerViewOnClickListener != null)
            this.recyclerViewOnClickListener.onLongPressClickListener(v, layoutPosition)
        return false
    }

    fun setFavorito(favorito: Favorito) {
        txtSentido.text = DestinoEnum.getSiglaByCodigo(favorito.codigoSentido)
        txtLinha.text = favorito.nomeLinha
    }
}
